pkg/config/config_template.yaml

```diff
@@ -1526,6 +1526,14 @@ api_key:
 #
 # dogstatsd_origin_detection: false
 
+## @param dogstatsd_origin_detection_type - string - optional - default: ""
+## @env DD_DOGSTATSD_ORIGIN_DETECTION_TYPE - string - optional - default: ""
+## A DogStatsD client can send its container ID.
+## Set this parameter to "client" to instruct the Agent to use the container ID to enrich the metrics with container tags.
```

Comment:
Are we planning other value than `client` ? Could this be a boolean instead ?

Reply:
@clamoriniere any thoughts on this?

Reply:
My main issue is that `dogstatsd_origin_detection_type: client` might imply that we will no longer support UDS origin detection while we would support both at the same time with the client side having priority.

Naming is always hard...
What about `dogstatsd_client_side_origin_detection: <bool>` ?

Or we keep the naming but make it a list of and deprecated `dogstatsd_origin_detection` ?
```
dogstatsd_origin_detection_type:
  - uds
  - client
  - ...
```
This will mean more code change in this PR and I'm not sure it's worth it.

WDYT ?

Reply:
Booleans are usually less error prone, so I'm leaning towards keeping it a bool - I renamed it `dogstatsd_origin_detection_client` in this [commit](https://github.com/DataDog/datadog-agent/pull/10659/commits/6148df977dd6c00573701c9d7c409c2774b6d7c1) . I can make it `dogstatsd_client_side_origin_detection` but it's a bit long